lib/libc/string/wcscmp.c

```diff
@@ -51,8 +51,8 @@ wcscmp(const wchar_t *s1, const wchar_t *s2)
 {
 
 	while (*s1 == *s2++)
-		if (*s1++ == '\0')
+		if (*s1++ == L'\0')
 			return (0);
-	/* XXX assumes wchar_t = int */
-	return (*(const unsigned int *)s1 - *(const unsigned int *)--s2);
+
+	return (int)((unsigned int)*s1 - (unsigned int)(*(s2 - 1)));
```

Comment:
The --s2 is stupid code but better to leave it as is for this change, especially if you're not changing wcsncmp to match

Reply:
Also style(9) requires a set of parens around this whole thing.